compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/AssertionLowering.kt

```diff
@@ -77,39 +76,23 @@ private class AssertionLowering(private val context: JvmBackendContext) :
         if (!function.isAssert)
             return super.visitCall(expression, data)
 
-        val mode = context.state.assertionsMode
-        if (mode == JVMAssertionsMode.ALWAYS_DISABLE)
-            return IrCompositeImpl(expression.startOffset, expression.endOffset, context.irBuiltIns.unitType)
-
-        context.createIrBuilder(expression.symbol).run {
-            at(expression)
-            val assertCondition = expression.getValueArgument(0)!!
-            val lambdaArgument = if (function.valueParameters.size == 2) expression.getValueArgument(1) else null
-
-            return if (mode == JVMAssertionsMode.ALWAYS_ENABLE) {
-                checkAssertion(assertCondition, lambdaArgument)
-            } else {
-                require(mode == JVMAssertionsMode.JVM && data != null)
-                irIfThen(
-                    irNot(getAssertionDisabled(this, data)),
-                    checkAssertion(assertCondition, lambdaArgument)
-                )
+        val constructor = context.ir.symbols.assertionErrorConstructor
```

Comment:
How are these changes related to KT-47483?

Reply:
Hello